src/main/java/org/folio/innreach/batch/contribution/IterationEventReaderFactory.java

```diff
@@ -44,12 +49,25 @@ public KafkaItemReader<String, InstanceIterationEvent> createReader(String tenan
     var topic = String.format("%s.%s.%s",
       folioEnv.getEnvironment(), tenantId, jobProperties.getReaderTopic());
 
-    var reader = new KafkaItemReader<String, InstanceIterationEvent>(props, topic);
+    var reader = new KafkaItemReader<>(props, topic, keyDeserializer(), valueDeserializer());
     reader.setPollTimeout(Duration.ofSeconds(jobProperties.getReaderPollTimeoutSec()));
     reader.setRecordProcessor(CONSUMER_REC_PROCESSOR);
+
     return reader;
   }
 
+  private Deserializer<InstanceIterationEvent> valueDeserializer() {
+    JsonDeserializer<InstanceIterationEvent> deserializer = new JsonDeserializer<>(InstanceIterationEvent.class, mapper);
+    deserializer.setUseTypeHeaders(false);
+    deserializer.addTrustedPackages("*");
```

Comment:
I guess this property is not needed when target class is explicitly specified in constructor